ksqldb-parser/src/main/java/io/confluent/ksql/util/ParserUtil.java

```diff
@@ -211,6 +211,16 @@ public static Optional<NodeLocation> getLocation(final Token token) {
     return Optional.of(new NodeLocation(token.getLine(), token.getCharPositionInLine()));
   }
 
+  public static Optional<NodeLocation> getEndLocation(final ParserRuleContext parserRuleContext) {
+    requireNonNull(parserRuleContext, "parserRuleContext is null");
+    return getEndLocation(parserRuleContext.getStop());
+  }
+
+  private static Optional<NodeLocation> getEndLocation(final Token stop) {
+    return Optional.of(new NodeLocation(stop.getLine(),
```

Comment:
can we add a comment here explaining the math (what are each of these and why are we adding stop and subtracting start and then adding 1?)

Reply:
sorry, this was still a draft. should have marked it as one as well @agavra